wade-review.md

```diff
@@ -0,0 +1,596 @@
+                        GitHubPRStateCounter stateCounter = kvStore.get(STORE_KEY);
+                        if (stateCounter == null) {
+                            stateCounter = new GitHubPRStateCounter();
+                        }
+                        if (prInfo.state().equalsIgnoreCase("open")) {
+                            stateCounter.setOpen(stateCounter.getOpen() + 1);
+                        } else {
+                            stateCounter.setClosed(stateCounter.getClosed() + 1);
+                        }
+                        kvStore.put(STORE_KEY, stateCounter);
+                    }
+                }
+            };
+        }
+
+        @Override
+        public Set<StoreBuilder<?>> stores() {
+            return Collections.singleton(Stores.keyValueStoreBuilder(Stores.inMemoryKeyValueStore(STORE_NAME), Serdes.String(), prStateCounterSerde));
+        }
+    }
+
+}
+```
+
+Lastly, we use the processor API to create our state store. We return a singleton with the store values built from our store builder. 
+
+
+## The GitHub PR Ratio class
+
+```java
+public class GitHubPrRatio {
+}
+```
+
+ Next, I’m going to create a GitHubPRRatio class to hold this processing… process!. I’ll need some variables:
+
+
+```java
+public class GitHubPrRatio {
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubPrRatio.class);
+
+}
+```
+A logger for verifying that the data’s coming in, 
+```java
+public class GitHubPrRatio {
+    // start with public class, then fill in variables in screen recording
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubPrRatio.class);
+    static final Serde<GitHubPRStateCounter> prStateCounterSerde = StreamsSerde.serdeFor(GitHubPRStateCounter.class);
+    final Serializer<JsonNode> jsonSerializer = new JsonSerializer();
+    final Deserializer<JsonNode> jsonDeserializer = new JsonDeserializer();
+    final Serde<JsonNode> jsonNodeSerde = Serdes.serdeFrom(jsonSerializer, jsonDeserializer);
+}
+```
+Variables working towards creating my custom serde,
+```java
+public class GitHubPrRatio {
+    // start with public class, then fill in variables in screen recording
+    private static final Logger LOG = LoggerFactory.getLogger(GitHubPrRatio.class);
+    static final Serde<GitHubPRStateCounter> prStateCounterSerde = StreamsSerde.serdeFor(GitHubPRStateCounter.class);
+    final Serializer<JsonNode> jsonSerializer = new JsonSerializer();
+    final Deserializer<JsonNode> jsonDeserializer = new JsonDeserializer();
+    final Serde<JsonNode> jsonNodeSerde = Serdes.serdeFrom(jsonSerializer, jsonDeserializer);
+    static final String STORE_NAME = "pr-state";
+    static final String NO_STATE = "no-state";
+    static final String INPUT_TOPIC = "github-pull_requests";
+    static final String OUTPUT_TOPIC = "github-pull_requests"
+
+}
+```
+And some strings – my storename, a no state variable, and some input and output topic names. 
```

Comment:
`INPUT_TOPIC` and `OUTPUT_TOPIC` are pretty clear here. I wouldn't mind a bit of an explanation of `NO_STATE`.  I can guess what `STORE_NAME` will be used for, but some explanation of the fact that we are going to need to create a store of some kind would help.

Reply:
https://github.com/Cerchie/notes-for-gh-video/pull/2/commits/181279df70a7cdb788a5c7fae754049966765849